src/providers/npm.py

```diff
@@ -0,0 +1,82 @@
+from typing import List, Callable, Tuple
+import requests
+import urllib.request
+import os
+import multiprocessing
+import itertools
+from semver import max_satisfying
+
+from src.providers.Provider import Provider
+
+NPM_REGISTRY_URL = 'https://registry.npmjs.org/'
+
+
+def _clean_package_version(version: str):
+    if version == '*' or version is None or version == 'latest':
+        return 'latest'
+    else:
+        return version.replace('~', '').replace('^', '')
+
+
+def _get_version_package_payload(package_name: str, version: str) -> dict:
+    response = requests.get(NPM_REGISTRY_URL + package_name)
+    if response.status_code == 404:
+        raise Exception('Module not found {0}:{1}'.format(package_name, version))
+    if response.status_code != 200:
+        raise Exception('Unknown error occurred {0}:{1}'.format(package_name, version))
+    response_payload = response.json()
+    if version == 'latest':
+        version = response_payload['dist-tags']['latest']
+    if version not in response_payload['versions']:
+        satisfied_version = max_satisfying(list(response_payload['versions'].keys()), version, loose=False)
+        if version is None:
+            print('Failed to satisfy {0}:{1}'.format(package_name, version))
+            raise Exception('Version was not found for {0}:{1}'.format(package_name, version))
+        print('instead of {0}:{1} Used {0}:{2}'.format(package_name, version, satisfied_version))
+        version = satisfied_version
+    return response_payload['versions'][version]
+
+
+def _get_deps(package_name: str, version: str, should_download_dev_deps=False) -> List[Tuple[str, str, dict]]:
```

Comment:
```suggestion
def _get_deps(package_name: str, version: str, should_download_dev_deps=False) -> List[Tuple[str, str, dict]]:
    version_response_payload = _get_version_package_payload(package_name, version)
    deps = []
    if 'dependencies' in version_response_payload:
        deps += version_response_payload['dependencies'].items()
    if should_download_dev_deps and 'devDependencies' in version_response_payload:
        deps += version_response_payload['devDependencies'].items()]
    return [(pkg_name, ver, _get_version_package_payload(pkg_name, _clean_package_version(ver))) for pkg_name, ver in joined_deps]
```

Reply:
it is unclear what is the change you would like me to make

Reply:
Go over the suggestion, it simplifies the function and avoids redundancies and code duplications

Reply:
done